### DISCUSSION TITLE
To play gif

### DISCUSSION THREAD

**Discussion by ApurvaSolanki54** [fltk-rs/fltk-rs#608] [Q&A]
Hey there!
I want to play a gif for my project. I have written the following code, it runs but doesn't play the gif properly.
I am writing the code below for you to refer

use std::env;
use std::fs::File;
use std::path::PathBuf;
use std::{thread, time};
use fltk::{app::*,button::*,window::*,input::*,dialog::*,frame::*,image::*,group::*,text::*,tree::*,draw};
fn main() -> Result<(), Box<dyn std::error::Error>> {
    // let input_path = PathBuf::from(
    //     env::args_os()
    //         .nth(1)
    //         .ok_or("Prints Error message")?,
    // );
    let main_app = App::default().with_scheme(Scheme::Gleam);
    let input = File::open("proj.gif")?;
    let mut options = gif::DecodeOptions::new();
    options.set_color_output(gif::ColorOutput::Indexed);
    let mut decoder = options.read_info(input)?;
    let screen_width = decoder.width();
    let screen_height = decoder.height();
    let global_pal = decoder.global_palette().unwrap_or_default().to_vec();

    let output_file_stem = "proj";
    let mut frame_number = 1;
    let mut wind = Window::default().with_size(500,500).center_screen();

    let mut frame_1 = Frame::default().size_of(&wind);
    // let mut img = GifImage::load("proj.gif").unwrap();
    // frame_1.set_image(Some(img));
    wind.end();
    wind.show();
    
    while main_app.wait() {

        while let Some(frame) = decoder.read_next_frame()? {
            let output_path = format!("{}.{:03}.gif", output_file_stem, frame_number);
            let mut output = File::create(&output_path)?;
            
            let mut encoder = gif::Encoder::new(&mut output, screen_width, screen_height, &global_pal)?;
            encoder.write_frame(&frame)?;
            frame_number += 1;

            use gif::DisposalMethod::*;
            let disposal = match frame.dispose {
                Any => "any",
                Keep => "keep",
                Background => "background",
                Previous => "previous",
            };
            let mut img = GifImage::load(&output_path).unwrap();
            frame_1.set_image(Some(img));
            wind.redraw();
            thread::sleep(time::Duration::from_millis(100));
            eprintln!(
                "Written {} ({}x{}@{}x{} delay={} {})",
                output_path, frame.width, frame.height, frame.top, frame.left, frame.delay, disposal
            );
        }
    }
    Ok(())
}

Can you help me by telling what can I possibly do?

**Comment by MoAlyousef** [comment]
Hi
Yeah the gif crate has issues rendering properly. Check here:
https://github.com/kornelski/image-gif-dispose
